### DISCUSSION TITLE
Proposed design for preserving both ownership and revenue balances

### DISCUSSION THREAD

**Discussion by revelator-labs** [originalworks/protocol#17] [Ideas]
### Intro

Perhaps we can explore a method where we change the underlying of `Tokenization` process of a `license` is done. Instead of creating a fractionalized EVM contract  each `RightsHolder` is issued a single EVM token to prove they are rights holders but don't expose their holdings... Then we mint their shares on a privacy preserving (non-EVM) blockchain.

Then, instead of paying the `RightsHolders` based on EVM-tokenized "splits" from the publicly available pool, we allow each user to anonymously batch-claim their funds to an "unknown" wallet. The only public information then will be the number of unique addresses owning any number of shares of the license, but each claim will be a pretty random collection of fractional earnings to a random address.

### How it could work

**User flow**

As we are planning to pay all royalties from one big pool that holds all the funds of that `PaymentProvider` then at the end of each month, rights holders can log in to the Registry or `PaymentProvider`'s custom application by connecting their publicly exposed (yet pseudonymous) EVM wallet that holds their single EVM token per track, representing their rights to a certain portion of it's cashflow. They can then claim all of their aggregate USDC royalties into a **_different_** EVM wallet - one with no Royalty tokens at all for example. Users can enter a different 0x address every month to further obfuscate their earnings.

**Execution**

The protocol or `PaymentpPovider` will need to identify the EVM tokens in the connected wallet, then read their privately-stored holdings from the non-EVM chain and send the USDC to any wallet that the token holder requests as a single batch payment. 
The payment provider should then submit their royalty reports through a private channel (maybe simply by using Whisper, native to the EVM stack) separate to the payment 

**This is a huge refactor to the wallet** which will need to be refactored anyway because of how we are planning to pool all royalty payments of each provider. 

- The Wallet will no longer hold "tokenized royalties" but just "keys" to view/move privately stored tokens and in addition it can also continue to be used as an EVM wallet, but only with token gating you will be able to access the additional private layer we will need to custom-build. 

- Other wallets will need the registry to move private shares around.

- We would essentially need (on both the registry and the wallet) to develop a system that can validate token holders and then (I) sign messages on a 3rd party blockchain in order to move their tokens around and (ii) access their private holdings to assign their royalty splits from their `PaymentProvider`'s  EVM royalty pool.

**Comment by revelator-labs** [comment]
If this method is an approach worth considering, then we need to also consider if this should be a process executed by the protocol or by the payment provider

**Comment by revelator-labs** [comment]
One of the options to explore could be (a potential EVM compatible solution?) https://minaprotocol.com

Their mainnet is live and prioritized cross-chain operability.

**Reply by revelator-labs** [reply]
another approach would be a private EVM chain holding all of the balances that only accepts transaction messages from key/token holders. Holders of the Royalty Tokens will be able to privately setup their private payment address so there will be no link between the token holding address and the royalty recipient address

**Comment by criadoperez** [comment]
This a significant shift in how tokenization is approached, moving from asset-based to user-based tokenization

In this case, how would then a user transfer part of his royalty tokens if he longer holds the balance of them?

Also, if it's user based tokenization, do we really need a token? If you go on that approach why not store on the private side what address on the public side stores what asset?

The concept of batch-claiming of royalties does add a layer of pseudonymity, which is nice.

If ownership and revenues balance must be private then we should consider the combination of permissioned and permissionless blockchains working in tandem. It will add complexity and costs to the protocol, but its an option.

**Reply by revelator-labs** [reply]
I agree this is a fundamental shift in design but I would argue it is keeping the tokenization approach (on the asset level) - just outsourcing it to a private or a privacy-preaerving public Blockchain.

But you are indeed right it adds a layer of user level control to the EVM stack that needs to govern messaging to the private Blockchain.
